DOC-12088: Fix the JavaScript UDF guides (#608)
couchbaselabs/docs-devex: modules/guides/pages/create-javascript-library.adoc

A scoped library is only available to users who have access to that bucket and scope.
Use a scoped JavaScript library to keep the code for user-defined functions separate.

The name of a JavaScript library must be unique within the specified namespace or scope.

include::partial$before-you-begin.adoc[tag=body]

[#creating-the-library-and-adding-your-first-function]
== Creating a JavaScript Library and Adding JavaScript Code

When you create a JavaScript library, you can add JavaScript functions to the library at the same time.

You can create a JavaScript library with the Couchbase Web Console or a REST API call.

[tabs]
====
Couchbase Web Console::
+
--
To create a JavaScript library:

. In the Couchbase Web Console, go to menu:Query[UDF].

. Under *JavaScript Function Libraries*, click *{plus}{nbsp}add function library*.
The *Add Library* dialog is displayed.

. In the *Namespace* drop-down list, select *(global)* for a global library, or select a bucket and scope for a scoped library.

. In the *Library Name* box, enter a name for the library.

. Specify *Parameters* for the function.

> . Edit the library to add your own JavaScript functions.

Author: The whole page is about libraries, but this particular step is about functions: it specifically says, "for the function". So in this case, parameters is correct.

. Click btn:[Save].

'''

The following library contains a JavaScript function called `getBusinessDays`.

[source.no-callouts, javascript]
----
include::javascript-udfs:example$get-business-days.js[]
----

--

REST API::
+
--
To create a JavaScript library with the Query Functions REST API:

. Use the xref:n1ql-rest-functions:index.adoc#post_library[`POST /evaluator/v1/libraries/{library}`] endpoint.

. Pass the name of the library as a path parameter.

. To create a scoped library, pass the bucket and scope as query parameters.

. Pass a string containing the JavaScript code for all functions in the library as the request body.

'''

The following request creates a JavaScript library in the `inventory` scope within the `travel-sample` bucket.

[source,sh]
----
include::example$javascript-udfs/create-scoped-javascript-function-with-rest.sh[]
----

--
====

> When you have created an external library and added JavaScript code, you must create an {sqlpp} user-defined function to reference the JavaScript code in the library, so it can be called as part of any {sqlpp} statement.

Reviewer: ```suggestion
After you create an external library and add JavaScript code, you must create a {sqlpp} user-defined function to reference the JavaScript code in the library, so it can be called as part of any {sqlpp} statement.
```
